src/prank_parallel.cpp

```diff
@@ -1,21 +1,23 @@
 #include <iostream>
 #include <fstream>
 #include <list>
+#include <vector>
 #include <unordered_map>
 #include <omp.h>
 #include <time.h>
 
 using namespace std;
 
 struct Page {
-    list <int> in_ids; //ids of pages that point to this page
+    //SP: should we keep this as vector or as a list ? is there be a specific order of Ids that can better the run time ?
```

Comment:
I'm not positive about c++ implementations, but I suspect vector is far superior for our purposes due to a single round of inserts and many lookups.

(I'm assuming underlying linked-list vs array, but they may both be array in which case it shouldn't matter)

Reply:
Ordering shouldn't matter too much because the graphs are sparse and so total edge count should be low and spread out.

Reply:
Hmm, page rank never does random access, but if the data is still located in contiguous memory locations, it would benefit us from cache perspective. So vector should perform better.